src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -181,29 +172,43 @@ public interface ISyntheticLocalReferenceExpression : IReferenceExpression
     /// </summary>
     public enum SyntheticLocalKind
     {
-        None,
+        None = 0x0,
 
         /// <summary>
         /// Created to capture the step value of a VB for loop.
         /// </summary>
-        ForLoopStepValue,
+        ForLoopStepValue = 0x1,
         /// <summary>
         /// Created to capture the limit value of a VB for loop.
         /// </summary>
-        ForLoopLimitValue
+        ForLoopLimitValue = 0x2
     }
 
     /// <summary>
-    /// Represents a reference to a C# this or VB Me parameter.
+    /// Represents a C# this or base expression, or a VB Me, MyClass, or MyBase expression.
     /// </summary>
-    public interface IInstanceReferenceExpression : IParameterReferenceExpression
+    public interface IInstanceReferenceExpression : IOperation
```

Comment:
I believe this should inherit from IReferenceExpression and not just IOperation since 'this' is effectively a byref argument for struct methods and can be assigned to.